test/Prompt/DontWrap.C

```diff
@@ -172,7 +172,7 @@ const Test2::Nested::Value::type & Test2::Nested::reference() const {
 Test2::Nested Nest2(80);
 // CHECK: Nested::Nested(160)
 Nest2.simpleAdd(3)
-// CHECK: (Test2::Nested::Value::type) 83
+// CHECK: (int) 83
```

Comment:
> Whats complicated is delineating between typedefs that are better desugared and those better left alone.

Yes and the 'underlying type' I was mentioning is a gateway to this.
In particular the output for

typedef int* intPtr_t;
intPtr_t val

should probably be 

(int *) @)0x...

rather than

(intPtr_t) 0x....

For the case, you mentioned the first list (just below 'From this thread I have another commit that does this:') long close to the better we can expect/do reasonably, I think ...